Lesson8/main.js

```diff
@@ -0,0 +1,20 @@
+const addLoading = () =>{
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img"
+  loading.appendChild(loadingImg)
+}
+
+const promiseReject = new Promise((resolve, reject) => {
+  addLoading();
+  setTimeout(() => {
+    reject("The promise was rejected!");
+  }, 3000);
+});
+
+promiseReject.then((error)　=> {
+})
+.catch((error) =>{
+  console.log(error)
+})
```

Comment:
IMO
If you want to handle errors in then, 
specify it as the second argument. However, 
since it also displays the error in catch, I think it will display the error twice.

This is an easy to understand article about Promise.
https://knowledge.sakura.ad.jp/24890/
```suggestion
promiseReject.then(
  (value) => {
    // 成功　esolve()に渡した値が引数として渡される
    console.log("OK", value);
  },
  (err) => {
    // 失敗　reject()に渡した値が引数として渡される
    console.log("error", err);
  }
).catch((error) => {　//　エラーをキャッチ
  console.log(error);
});
```

Reply:
Thank you so much for sharing the article, it was vert informative! I would take time to understand the concept of the article, but I will come back to it for reference.
Thank you for showing me the example codes, which was so helpful to understand the different ways to handle a reject process using then() and catch() in promise.

Please check the commit b4df1a91845e32070e5138a90d080cd2f28b5993